src/databricks/labs/ucx/hive_metastore/views_migrator.py

```diff
@@ -0,0 +1,105 @@
+import sqlglot
+from sqlglot import ParseError
+from sqlglot.expressions import Expression as SqlExpression
+from sqlglot.expressions import Table as SqlTable
+
+from databricks.labs.ucx.hive_metastore import TablesCrawler
+from databricks.labs.ucx.hive_metastore.tables import Table
+
+
+class ViewToMigrate:
+
+    table: Table
+    table_dependencies: set[Table]
+    view_dependencies: set[Table]
+
+    def __init__(self, table: Table):
+        if table.view_text is None:
+            raise RuntimeError("Should never get there! A view must have 'view_text'!")
+        self.table = table
+        self.table_dependencies = set()
+        self.view_dependencies = set()
+
+    def compute_dependencies(self, all_tables: dict[str, Table]):
+        if len(self.table_dependencies) + len(self.view_dependencies) == 0:
+            statement = self._parse_view_text()
+            for sql_table in statement.find_all(SqlTable):
+                catalog = self._catalog(sql_table)
+                if catalog != 'hive_metastore':
+                    continue
+                table_with_key = Table(catalog, sql_table.db, sql_table.name, "type", "")
+                table = all_tables.get(table_with_key.key)
+                if table is None:
+                    raise ValueError(
+                        f"Unknown schema object: {table_with_key.key} in view SQL: {self.table.view_text} of table {self.table.key}"
+                    )
+                if table.view_text is None:
+                    self.table_dependencies.add(table)
+                else:
+                    self.view_dependencies.add(table)
+
+    def _parse_view_text(self) -> SqlExpression:
+        try:
+            # below can never happen but avoids a pylint error
+            assert self.table.view_text is not None
+            statements = sqlglot.parse(self.table.view_text)
+            if len(statements) != 1 or statements[0] is None:
+                raise ValueError(f"Could not analyze view SQL: {self.table.view_text} of table {self.table.key}")
+            return statements[0]
+        except ParseError as e:
+            raise ValueError(f"Could not analyze view SQL: {self.table.view_text} of table {self.table.key}") from e
+
+    # duplicated from FromTable._catalog, not sure if it's worth factorizing
+    @staticmethod
+    def _catalog(table):
+        if table.catalog:
+            return table.catalog
+        return 'hive_metastore'
+
+    def __hash__(self):
+        return hash(self.table)
+
+
+class ViewsMigrator:
+
+    def __init__(self, crawler: TablesCrawler):
+        self.crawler = crawler
+        self.result_view_list: list[ViewToMigrate] = []
+        self.result_tables_set: set[Table] = set()
+
+    def sequence(self) -> list[Table]:
+        # sequencing is achieved using a very simple algorithm:
```

Comment:
that's a neat algorithm. shouln't we integrate it with `Index` here to line up only the views with only migrated dependencies?..

Reply:
Possibly, but I don't yet know enough details about the migration integration to provide an opinion. Happy to add that suggestion to a ticket.

Reply:
@nfx , it's not neat, can improve a lot. Please have a look at my below comment.